### PR TITLE
Docs and Requests for Create Binding

### PR DESCRIPTION
**Pull request by ralikio**
**Description**

Documentation for Kyma bindings.

Changes proposed in this pull request:

- `05-60-kyma-bindings.md` doc,
- new `requests` folder in the project root,
- example http requests for binding creation.

**Related issue(s)**

#284

### CODE REVIEW

File: docs/user/05-60-kyma-bindings.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Kyma Bindings
+
+One of the Broker API endpoints related to bindings allow for management of credentials for accessing given service. The endpoints in question include all subpaths of `v2/service_instances/<service_id>/service_bindings`. In case of Kyma Environment Broker, the generated credentials are in the form of a Kubeconfig for managed SKR cluster. To generated a kubeconfig for a given SKR instance sent the following request to the Broker API:
```
Comment (IwonaLanger) [inline docs/user/05-60-kyma-bindings.md:3]:
```suggestion
A Broker API endpoint related to bindings allows for managing credentials for accessing a given service. The Broker API endpoints include all subpaths of `v2/service_instances/<service_id>/service_bindings`. In the case of Kyma Environment Broker (KEB), the generated credentials are a kubeconfig for the managed Kyma cluster. To generate a kubeconfig for a given Kyma instance, send the following request to the Broker API:
```

File: docs/user/05-60-kyma-bindings.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+{
+  "service_id": "{{service_id}}",
+  "plan_id": "{{plan_id}}"
+  "parameters": {
+  }
+}
+```
+
+As a result of the above call the Broker will return a Kubeconfig file in the response body. The Kubeconfig file will contain the necessary information to access the managed SKR cluster. By default, Kyma Environment Broker uses [`shoots/adminkubeconfig`](https://github.com/gardener/gardener/blob/master/docs/usage/shoot_access.md#shootsadminkubeconfig-subresource) subresources to generate a Kubeconfig that uses certificates to authenticate its user. To customize format of returned kubeconfig the following parameters (used in the `parameters` section of the request body) can be used:
```
Comment (IwonaLanger) [inline docs/user/05-60-kyma-bindings.md:18]:
```suggestion
The Broker returns a kubeconfig file in the response body. The kubeconfig file contains the necessary information to access the managed Kyma cluster. By default, KEB uses [`shoots/adminkubeconfig`](https://github.com/gardener/gardener/blob/master/docs/usage/shoot_access.md#shootsadminkubeconfig-subresource) subresources to generate a kubeconfig that uses certificates to authenticate its user. To customize the format of the returned kubeconfig, use the same parameters that were used in the `parameters` section of the request body:
```

File: docs/user/05-60-kyma-bindings.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+| Name | Default | Description |
+|---|---|---|
+| `token_request` | `false` | If set to `true` the Broker will return a kubeconfig with JWT token used as user authentication mechanism. The token itself ise generated using K8S's TokenRequest that is attached to a ServiceAccount, ClusterRole and ClusterRoleBinding all named `kyma-binding-{{binding_id}}`. Such approach allows to easily modify permissions granted to the Kubeconfig. |
```
Comment (IwonaLanger) [inline docs/user/05-60-kyma-bindings.md:22]:
```suggestion
| **token_request** | `false` | If set to `true`, the Broker returns a kubeconfig with a JWT token used as a user authentication mechanism. The token is generated using Kubernetes TokenRequest attached to a ServiceAccount, ClusterRole, and ClusterRoleBinding, all named `kyma-binding-{{binding_id}}`. Such an approach allows for easily modifying the permissions granted to the kubeconfig. |
```

File: requests/bindings/create-tokenrequest.http [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+PUT http://localhost:8080/oauth/v2/service_instances/{{instance_id}}/service_bindings/{{binding_id}}?accepts_incomplete=true&accepts_incomplete=true&service_id={{service_id}}&plan_id={{plan_id}}
+Content-Type: application/json
+X-Broker-API-Version: 2.14
+
+{
+  "service_id": "{{service_id}}",
+  "plan_id": "{{plan_id}}",
+  "parameters": {
+    "token_requests": false
```
Comment (jaroslaw-pieszka) [inline requests/bindings/create-tokenrequest.http:17]:
```suggestion
    "token_request": false
```

File: docs/user/02-10-broker-endpoints.md [hunk 5]
```diff
@@ -12,4 +12,4 @@ Kyma Environment Broker (KEB) implements the [Open Service Broker API (OSB API)]
 
 Besides OSB API endpoints, KEB exposes the REST `/info/runtimes` endpoint that provides information about all created Runtimes, both succeeded and failed. This endpoint is secured with the OAuth2 authorization.
 
-For more details on KEB APIs, see [this file](../../files/swagger/index.html).
+For more details on KEB APIs, see [this file](../../files/swagger/index.html).
```
Comment (IwonaLanger) [inline docs/user/02-10-broker-endpoints.md:15]:
```suggestion
For more details on KEB APIs, see the [`swagger`](../../resources/keb/files/swagger.yaml) file.
```

Comment (IwonaLanger) [reply docs/user/02-10-broker-endpoints.md:15]: It's been agreed to point to the YAML file here.

File: docs/user/05-60-kyma-bindings.md (hunk 2 above)
Comment (ralikio) [reply docs/user/05-60-kyma-bindings.md]: Corrected.

**Review by IwonaLanger — APPROVED** [review]

**Review by jaroslaw-pieszka — APPROVED** [review]

**Review by IwonaLanger — APPROVED** [review]

**Pull request merged**